community/cypher/acceptance/src/test/scala/org/neo4j/internal/cypher/acceptance/PreParsingAcceptanceTest.scala

```diff
@@ -0,0 +1,210 @@
+/*
+ * Copyright (c) 2002-2015 "Neo Technology,"
+ * Network Engine for Objects in Lund AB [http://neotechnology.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Neo4j is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.neo4j.internal.cypher.acceptance
+
+import org.neo4j.cypher.internal.compatibility.{CompatibilityPlanDescription, CompatibilityPlanDescriptionFor2_3}
+import org.neo4j.cypher.internal.compiler.v2_3._
+import org.neo4j.cypher.internal.compiler.v2_2
+import org.neo4j.cypher.{ExecutionEngineFunSuite, ExtendedExecutionResult}
+import org.scalatest.matchers.{MatchResult, Matcher}
+
+class PreParsingAcceptanceTest extends ExecutionEngineFunSuite {
```

Comment:
Ah, damn, we didn't find that one. And looking at it, actually, there's a small difference in test semantics. The `CypherPreParserTest` class is a unit test, _only_ testing the output of `parse`. What we've written now is a proper acceptance test, that actually runs the queries and checks the planner used only at the end.

Do you feel strongly about this, or do you think it could make sense to have both perhaps?